src/pages/Home.jsx

```diff
@@ -1,12 +1,73 @@
 import React from "react";
+import dataJson from "../assets/CountriesData.json"
+import { useState } from "react";
+import Country from "../components/Country"
+import Header from "../components/Header"
+import Filters from "../components/Filters"
+
+
 
 const Home = () => {
+    const [filteredArray, setfilteredArray] = useState(dataJson);
+    const [isDarkMode, setIsDarkMode] = useState(false); 
+    const filterByRegion = (region) => {
+        console.log(region);
+        
+        if(region==="All"){
+            setfilteredArray(dataJson);
+        }
+        else{
+            const filtered = dataJson.filter((Country)=>
+                Country.region.toLowerCase() === region.toLowerCase());
+            setfilteredArray(filtered)
+        }
+    };
+
+    const matchValue = (country,input) =>{
+        return country.name.toLowerCase().includes(input) ||
+            country.population.toString().includes(input) ||
+            country.region.toLowerCase().includes(input) ||
```

Comment:
includes is array function so you could do it like this :

```js
const matchValue = (country, input) => 
  [country.name, country.population.toString(), country.region, country.capital]
    .some(value => value.toLowerCase().includes(input.toLowerCase()));

```

And if you destruct card it will be even shorter